### PR TITLE
evcxr: 0.3.3 -> 0.4.3

### PR DESCRIPTION
**Pull request by Ma27**
###### Motivation for this change

The current version fails to build: https://hydra.nixos.org/build/99189245

Please note that the releases aren't tracked by Git, but the currently
used rev points to `0.4.3` at crates.io: https://crates.io/crates/evcxr/0.4.3

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @protoben

### CODE REVIEW

File: pkgs/development/interpreters/evcxr/default.nix [hunk 1]
```diff
@@ -1,21 +1,20 @@
...
     owner = "google";
     repo = "evcxr";
-    rev = "ae07ccf08723b7aec0de57d540822b89088ca036";
-    sha256 = "1apc93z9vvf6qks5x2pad45rnrj9kjl812rj78w5zmmizccp2fhf";
+    rev = "58ef3c7914ab0d958144fe8615a10c9653b217d0";
+    sha256 = "08zsdl6pkg6dx7k0ns8cd070v7ydsxkscd2ms8wh9r68c08vwzcp";
   };
 
-  cargoSha256 = "153pxqj4jhlbacr7607q9yfw6h96ns5igbvssis8j3gn0xp6ssg6";
-  cargoPatches = [ ./cargo-lock.patch ];
```
Comment (worldofpeace) [inline pkgs/development/interpreters/evcxr/default.nix]: Did you delete the patch in tree?

**Comment by worldofpeace** [comment]
Opened issue for tags https://github.com/google/evcxr/issues/59

File: pkgs/development/interpreters/evcxr/default.nix [hunk 2]
```diff
@@ -1,21 +1,20 @@
...
+  cargoSha256 = "1hqlagwl94xcybfqq5h2mrz9296mjns2l598d6jclls7ac5wsdfc";
 
   nativeBuildInputs = [ pkgconfig makeWrapper ];
-  buildInputs = [ cargo ] ++ stdenv.lib.optional stdenv.isDarwin Security;
+  buildInputs = [ cargo zeromq ] ++ stdenv.lib.optional stdenv.isDarwin Security;
```
Comment (worldofpeace) [inline pkgs/development/interpreters/evcxr/default.nix]: buildRustPackage does
```
nativeBuildInputs = nativeBuildInputs ++ [ cacert git cargo rustc ];
```

is it needed in `buildInputs`?

File: pkgs/development/interpreters/evcxr/default.nix [hunk 3]
```diff
@@ -1,21 +1,20 @@
-{ cargo, fetchFromGitHub, makeWrapper, pkgconfig, rustPlatform, stdenv, gcc, Security }:
+{ cargo, fetchFromGitHub, makeWrapper, pkgconfig, rustPlatform, stdenv, gcc, Security, zeromq }:
 
 rustPlatform.buildRustPackage rec {
   name = "evcxr-${version}";
```
Comment (worldofpeace) [inline pkgs/development/interpreters/evcxr/default.nix]: pname?

**Review by worldofpeace — COMMENTED** [review]
Builds and executes on NixOS.

File: pkgs/development/interpreters/evcxr/default.nix [hunk 4]
```diff
@@ -1,21 +1,20 @@
...
-  version = "0.3.3";
+  version = "0.4.3";
 
   src = fetchFromGitHub {
     owner = "google";
     repo = "evcxr";
-    rev = "ae07ccf08723b7aec0de57d540822b89088ca036";
-    sha256 = "1apc93z9vvf6qks5x2pad45rnrj9kjl812rj78w5zmmizccp2fhf";
+    rev = "58ef3c7914ab0d958144fe8615a10c9653b217d0";
```
Comment (worldofpeace) [inline pkgs/development/interpreters/evcxr/default.nix]: Author just tagged all their releases https://github.com/google/evcxr/releases/tag/v0.4.3 :sparkles:

Comment (Ma27) [reply pkgs/development/interpreters/evcxr/default.nix]: Awesome! Thanks a lot for taking care of this! :)

File: pkgs/development/interpreters/evcxr/default.nix (hunk 2 above)
Comment (Ma27) [reply pkgs/development/interpreters/evcxr/default.nix]: After taking a closer look, I'm pretty sure that you are right. Currently updating the PR accordingly :)

**Review by worldofpeace — APPROVED** [review]
LGTM, still builds.

**Pull request merged**